src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/ConsensusRuleUnitTestBase.cs

```diff
@@ -197,7 +197,7 @@ public TestConsensusRulesUnitTestBase() : base(KnownNetworks.TestNet)
         public override TestConsensusRules InitializeConsensusRules()
         {
             return new TestConsensusRules(this.network, this.loggerFactory.Object, this.dateTimeProvider.Object, this.concurrentChain, this.nodeDeployments,
-                this.consensusSettings, this.checkpoints.Object, this.chainState.Object, new InvalidBlockHashStore(new DateTimeProvider()));
+                this.consensusSettings, this.checkpoints.Object, this.chainState.Object, new InvalidBlockHashStore(this.dateTimeProvider.Object), new NodeStats(new DateTimeProvider()));
```

Comment:
Shouldn't this also use `this.dateTimeProvider.Object` (last parameter)

Reply:
doesn't make any difference if we use new time provider or a mock. 
I personally prefer using actual instances whenever possible because it makes the test more robust.